src/main/java/ch/njol/skript/expressions/ExprIP.java

```diff
@@ -85,37 +122,40 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 				}
 				addr = socketAddr.getAddress();
 			}
-			
+
 			// Check if address is not available, just in case...
 			if (addr == null) {
 				ips[i] = "unknown";
 				continue;
 			}
-			
+
 			// Finally, place ip here to array...
 			ips[i] = addr.getHostAddress();
 		}
-		
+
 		return ips;
 	}
-	
-	@Override
-	public Class<String> getReturnType() {
-		return String.class;
-	}
 
 	@Override
 	public boolean isSingle() {
 		return true;
 	}
 
+	@Override
+	public Class<String> getReturnType() {
+		return String.class;
+	}
 
 	@Override
 	public String toString(@Nullable Event e, boolean debug) {
-		if (e != null)
-			return "ip of " + players.toString(e, debug);
-		else
+		if (e != null) {

```

Comment:
this seems wrong. you pretty much never need to do stuff with the event in toStrings, and you definitely don't need to here

Reply:
I'm not who added it, and 
![image](https://user-images.githubusercontent.com/19474570/44484637-f9043180-a656-11e8-990a-2e69b92d7cc7.png)
![image](https://user-images.githubusercontent.com/19474570/44484656-06212080-a657-11e8-8a28-f7257c91cf23.png)